### ISSUE TITLE
Posible error test case E3/019_Hybrid_Moon-Twin_wing

### ISSUE THREAD

**Issue by VicentePareja** [franciscogazitua/IIC2113-2024#60]
Hola!

En el test case E3/019_Hybrid_Moon-Twin_wing creo que podría haber un error. específicamente en le línea "Eirika reducirá el daño de los ataques del rival en un 36%" dado que creo "Eirika reducirá el daño de los ataques del rival en un 16%" es el output correcto. Esto producto que la diferencia de spd entre Eirika y Robin es de 4. 

Un saludo!

Vicente

**Comment by cahinostroza** [comment]
Hola!

Está bien, la diferencia normal es de 4, pero luego de aplicarse el penalty es de 9. `9 * 4 = 36`

Saludos

**Comment by RodrigoToroIcarte** [comment]
Efectivamente. Recuerda que ese efecto es sobre la diferencia de spds luego de aplicado los bonus, penalties y neutralizadores que afecten al combate completo.

**Comment by VicentePareja** [comment]
Gracias!

**Comment by FranciscoPinto1138** [comment]
Sobre este mismo test, ¿por qué no se imprime en el round 8 "Alm reducirá el daño de los ataques del rival en un 40%" si es que Alm tiene Spd: 51 y Ephraim Spd: 37? Gracias de antemano.

**Comment by VicentePareja** [comment]
Arreglé lo anterior y tengo la misma duda! Gracias de antemano!

**Comment by anibalcontreras** [comment]
Me sumo a la duda, tengo los mismos valores de Spd para Alm y Ephraim

**Comment by RodrigoToroIcarte** [comment]
Creo que es porque la condición "_si el HP de la unidad >= 25%_" no se cumple.

**Comment by anibalcontreras** [comment]
Respecto a eso, tengo que la unidad tiene HP = 13 y el Base HP es 57 -> Hp de la unidad es 22%. Asi que efectivamente no se cumple.

La descripción de la skill es: Al inicio del combate, si el HP de la unidad ≥ 25%, inflige Atk/Spd-5 en el rival durante el combate, y también, si la Spd de la unidad > Spd del rival, reduce el daño de los ataques del rival durante el combate en un porcentaje = diferencia entre los stats x 4 (máx. 40 %).

Esto lo estoy entendiendo como
Efecto 1) Si el HP de la unidad >= 25 -> inflingir Atk/Spd-5
Efecto 2) Si Spd de la Unidad > Spd del Rival -> Reducir el daño de los ataques del rival

Es decir, como el HP de la unidad < 25%, entonces no aplicamos el penalty, pero si se cumple que el spd de la unidad > al spd del rival. Entonces si se debe reducir el daño de los ataques del rival.

Si no es asi, como tendría que ser?

Gracias de antemano

**Comment by jpedrodelrio** [comment]
Hola! Me sumo a la misma duda, tengo los mismos valores de Spd para Alm y Ephraim

**Comment by franciscogazitua** [comment]
Hola

Las condiciones son inclusivas, por eso se incluye el `y también` en su descripción. Entonces:

1. Si el HP de la unidad >= 25%, entonces se aplica Atk/Spd-5
2. Si el HP de la unidad es >= 25% y Spd de la unidad > Spd del rival, entonces se reduce el daño

Saludos
